nums/core/array/blockarray.py

```diff
@@ -883,12 +883,26 @@ def _block_shape_reshape(self, arr, block_shape):
     def _strip_ones(self, shape):
         return tuple(filter(lambda x: x != 1, shape))
 
+    def _check_positions_ones(self, shape, block_shape):
+        for i in range(len(shape)):
+            if shape[i] == 1:
```

Comment:
Can you add a comment here explaining why we check for shape for 1 instead of block_shape?

Reply:
Will add.